src/LiveComponent/src/Twig/LiveComponentRuntime.php

```diff
@@ -24,51 +25,51 @@
  */
 final class LiveComponentRuntime
 {
-    private LiveComponentHydrator $hydrator;
-    private ComponentFactory $factory;
-    private UrlGeneratorInterface $urlGenerator;
-    private ?CsrfTokenManagerInterface $csrfTokenManager;
-
-    public function __construct(LiveComponentHydrator $hydrator, ComponentFactory $factory, UrlGeneratorInterface $urlGenerator, CsrfTokenManagerInterface $csrfTokenManager = null)
-    {
-        $this->hydrator = $hydrator;
-        $this->factory = $factory;
-        $this->urlGenerator = $urlGenerator;
-        $this->csrfTokenManager = $csrfTokenManager;
+    public function __construct(
+        private Environment $twig,
+        private LiveComponentHydrator $hydrator,
+        private ComponentFactory $factory,
+        private UrlGeneratorInterface $urlGenerator,
+        private ?CsrfTokenManagerInterface $csrfTokenManager = null
+    ) {
     }
 
-    public function renderLiveAttributes(Environment $env, object $component, string $name = null): string
+    public function renderLiveAttributes(array $context): string
     {
-        $name = $this->nameFor($component, $name);
-        $url = $this->urlGenerator->generate('live_component', ['component' => $name]);
-        $data = $this->hydrator->dehydrate($component);
-
-        $ret = sprintf(
-            'data-controller="live" data-live-url-value="%s" data-live-data-value="%s"',
-            twig_escape_filter($env, $url, 'html_attr'),
-            twig_escape_filter($env, json_encode($data, \JSON_THROW_ON_ERROR), 'html_attr'),
-        );
-
-        if (!$this->csrfTokenManager) {
-            return $ret;
+        if (!isset($context['_component_config'])) {
```

Comment:
Correct me if I'm wrong, but if we removed `renderLiveAttributes()`, then we would NOT need to include `_component_config` here, right? That would be a nice cleanup - it looks ugly and the `AddLiveAttributesSubscriber` does not need it.

Reply:
Yes, if we decide to remove `init_live_component()` completely (I'm in favor of this), I think we should include this logic in the subscriber itself.

Reply:
Let's remove it completely please :)